core/src/main/java/org/infinispan/interceptors/impl/NonTxIracLocalSiteInterceptor.java

```diff
@@ -194,7 +189,13 @@ private void visitKey(Object key, WriteCommand command) {
       if (getOwnership(segment) != Ownership.PRIMARY) {
          return;
       }
-      IracMetadata metadata = iracVersionGenerator.generateNewMetadata(segment);
+      IracMetadata metadata;
+      // RemoveExpired should lose to any other conflicting write
+      if (command instanceof RemoveExpiredCommand) {
+         metadata = iracVersionGenerator.generateMetadataWithCurrentVersion(segment);
```

Comment:
well, I was thinking about using the IracMetadata stored in the context entry.

Reply:
How would this be different sorry?